paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/PaymentSheetPlaygroundViewModel.kt

```diff
@@ -117,6 +137,112 @@ internal class PaymentSheetPlaygroundViewModel(
         }
     }
 
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    private fun prepareCustomer(
+        playgroundSettings: PlaygroundSettings
+    ) {
+        viewModelScope.launch {
+            val snapshot = playgroundSettings.snapshot()
+
+            snapshot.saveToSharedPreferences(getApplication())
+
+            state.value = PlaygroundState.Customer(
+                snapshot = snapshot,
+                adapter = CustomerAdapter.create(
+                    context = getApplication(),
+                    customerEphemeralKeyProvider = {
+                        fetchEphemeralKey(snapshot)
+                    },
+                    setupIntentClientSecretProvider = if (
+                        snapshot[CustomerSheetPaymentMethodModeDefinition] == PaymentMethodMode.SetupIntent
+                    ) {
+                        { customerId -> createSetupIntentClientSecret(snapshot, customerId) }
+                    } else {
+                        null
+                    }
+                )
+            )
+        }
+    }
+
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    private suspend fun fetchEphemeralKey(
+        snapshot: PlaygroundSettings.Snapshot
+    ): CustomerAdapter.Result<CustomerEphemeralKey> {
+        val requestBody = snapshot.customerEphemeralKeyRequest()
+
+        val apiResponse = Fuel.post(settings.playgroundBackendUrl + "customer_ephemeral_key")
+            .jsonBody(Json.encodeToString(CustomerEphemeralKeyRequest.serializer(), requestBody))
+            .suspendable()
+            .awaitModel(CustomerEphemeralKeyResponse.serializer())
+
+        return when (apiResponse) {
+            is Result.Failure -> {
+                val exception = apiResponse.getException()
+
+                CustomerAdapter.Result.failure(
+                    cause = exception,
+                    displayMessage = "Failed to fetch ephemeral key:\n${exception.message}"
+                )
+            }
+            is Result.Success -> {
+                val response = apiResponse.value
+
+                // Init PaymentConfiguration with the publishable key returned from the backend,
+                // which will be used on all Stripe API calls
+                PaymentConfiguration.init(
+                    getApplication(),
+                    response.publishableKey
+                )
+
+                try {
+                    CustomerAdapter.Result.success(
+                        CustomerEphemeralKey.create(
+                            customerId = response.customerId,
```

Comment:
Do we need to update the (playgrounds stored) customerId if the customer type is new?